src/epjson/epJSONTranslator.cpp

```diff
@@ -0,0 +1,452 @@
+                visitor(*fieldInt);
+                return true;
+              }
+            }
+
+            visitor(*fieldDouble);
+            return true;
+          }
+        }
+        case JSONValueType::Array:
+        case JSONValueType::Object:
+          break;
+      }
+
+      {
+        const auto fieldString = field.getString(idx);
+        if (fieldString && !fieldString->empty()) {
+          visitor(fixupEnumerationValue(schema, *fieldString, type_description, group_name, fieldName, iddField.properties().type));
+
+          return true;
+        }
+      }
+
+      return false;
+    };
+
+    std::size_t cur_group_number = 0;
+
+    for (const auto& g : obj.extensibleGroups()) {
+      ++cur_group_number;
+      // get first field and try to make a group name out of it
+      const auto& [group_name, is_array_group] =
+        openstudio::epJSON::getGroupName(group_names, field_names, schema, type_description, obj.iddObject().extensibleGroup()[0].name());
+
+      auto& containing_json = [&json_object, &group_name = group_name, is_array_group = is_array_group ]() -> auto& {
+        if (is_array_group) {
+          auto& array_obj = json_object[group_name];
+          return array_obj.append(Json::Value{Json::objectValue});
+        } else {
+          return json_object;
+        }
+      }
+      ();
+
+      for (unsigned int idx = 0; idx < g.numFields(); ++idx) {
+        const auto& iddField = obj.iddObject().extensibleGroup()[idx];
+
+        const auto fieldName = getFieldName(is_array_group, obj.iddObject(), schema, type_description, cur_group_number, idx,
+                                            toJSONFieldName(field_names, iddField.name()));
+
+        [[maybe_unused]] const auto fieldAdded = visitField([&containing_json, &fieldName](const auto& value) { containing_json[fieldName] = value; },
+                                                            iddField, group_name, fieldName, g, idx);
+      }
+    }
+
+    for (unsigned int idx = 0; idx < obj.numFields(); ++idx) {
+      const auto& iddField = obj.iddObject().getField(idx);
+
+      const auto& fieldName = toJSONFieldName(field_names, iddField->name());
+
+      if (iddField->isNameField()) {
+        // skip name, we already got that
+        continue;
+      }
+
+      if (obj.iddObject().isExtensibleField(idx)) {
+        // skip extensible field, we already dealt with that
+        continue;
+      }
+
+      visitField([&json_object, &fieldName](const auto& value) { json_object[fieldName] = value; }, iddField.get(), "", fieldName, obj, idx);
+    }
+  }
+  return result;
+}
+
+Json::Value toJSON(const openstudio::model::Model& model, const openstudio::path& schemaToLoad) {
+  openstudio::energyplus::ForwardTranslator trans;
+
+  openstudio::Workspace workspace = trans.translateModel(model);
+  workspace.toIdfFile();
+
+  return toJSON(workspace.toIdfFile(), schemaToLoad);
+}
```

Comment:
I agree with @kbenne this should be in openstudio::energyplus::forwardtranslator. Reading the method I assumed it would serialize my Model into JSON form, not translate to IDF then serialize that.

Reply:
Or another thought I had. should the translator be left alone and this work / feature accessed by IdfFile::saveAsJSON(). The wrinkle is that would only be possible if the IDD backing the IdfFile instance is the EnergyPlus idd. 

I'd appreciate some discussion on this topic, but also from my point of view this is just ironing out how we want to present the feature and not a question about the core capability and implementation.